backend/app/models/question_type.rb

```diff
@@ -3,10 +3,23 @@ class QuestionType < ApplicationRecord
   validates :name, uniqueness: { case_sensitive: false }
   has_many :questions, dependent: :destroy
 
+  before_destroy :check_questions_before_destroy, prepend: true
+
   def single_choice?
-    name == QuestionType::SINGLE_CHOICE
+    name == SINGLE_CHOICE
```

Comment:
I think you should keep this line the way it was and delete the constants `SINGLE_CHOICE` and `MULTIPLE_CHOICE` from this file, I think makes more sense to have them only on QuestionType. (having constants with the same name and same value in different files doesn't sound nice)

Reply:
but this is the question_type file?

Reply:
I must be crazy, my bad